packages/telemetry/src/service.ts

```diff
@@ -18,9 +18,23 @@
 import { _FirebaseService, FirebaseApp } from '@firebase/app';
 import { Telemetry } from './public-types';
 import { LoggerProvider } from '@opentelemetry/sdk-logs';
+import { _FirebaseInstallationsInternal } from '@firebase/installations';
 
 export class TelemetryService implements Telemetry, _FirebaseService {
-  constructor(public app: FirebaseApp, public loggerProvider: LoggerProvider) {}
+  fid?: string;
+
+  constructor(public app: FirebaseApp, public installationsProvider: _FirebaseInstallationsInternal, public loggerProvider: LoggerProvider) {
+    void this._getFid();
```

Comment:
@hsubox76 Due to the nature of our API, the `captureError` method needs to stay synchronous, but retrieving the `fid` is async. This leads to the `fid` being set here on subsequent invocations of `captureError`, not the first. Any ideas on how to have the `fid` in place in time for the first `captureError` call while keeping that method sync? cc @andrewbrook

Reply:
Updated the PR to do just that! Created a `DynamicLogAttributeProvider` which we can use in the future for other async log attributes 👍